### PR TITLE
Dhruv new develop

### PR DESCRIPTION
**Pull request by DhruvRathod77**
pull

### CODE REVIEW

File: obgyn/urls.py [hunk 1]
```diff
@@ -42,6 +42,9 @@
     path('patient/', include('patient.urls')),
     path('opd/', include('patient_opd.urls')),
     path('report/', include('reports.urls')),
+    path('subscription/', include('subscription.urls')),
     path('template-header/', include('template_header.urls')),
+    path('subscription_purchase/', include('subscription_purchase.urls')),
```
Comment (akhatri84) [inline obgyn/urls.py:47]: Check the patients.urls. 
include the subscription_purchase end point within subscription.urls. 
main urls.py should have minimum end points.

File: obgyn_config/models.py [hunk 2]
```diff
@@ -12,9 +18,13 @@ class ObgynConfigModel(models.Model):
     rs_per_room = models.FloatField(default=0, null=True)
     operative_charge = models.FloatField(default=0, null=True)
     rs_per_day_nursing = models.FloatField(default=0, null=True)
+    field_master_name  = models.ForeignKey(FieldMasterModel, on_delete=models.DO_NOTHING, null=True)
+    city_id = models.ForeignKey(CityModel, on_delete=models.CASCADE, null=True)
+    taluka_id = models.ForeignKey(TalukaModel, on_delete=models.CASCADE, null=True)
+    district_id = models.ForeignKey(DistrictModel, on_delete=models.CASCADE, null=True)
+    state_id = models.ForeignKey(StateModel, on_delete=models.CASCADE, null=True)
```
Comment (akhatri84) [inline obgyn_config/models.py:25]: Change every foreign key names. remove _id from foreign keys as it is not required.

File: obgyn_config/views.py [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+from utility.decorator import validate_permission, validate_permission_id
+
+
 from django.shortcuts import render
 from user.models import User
 from financial_year.models import FinancialYearModel
 from django.utils.timezone import now
 from patient_usgform.models import PatientUSGFormModel
-from obgyn_config.models import ObgynConfigModel
+from obgyn_config.models import ObgynConfigModel, FieldMasterModel
```
Comment (akhatri84) [inline obgyn_config/views.py]: Is FieldMasterModel import correct

File: obgyn_config/views.py [hunk 4]
```diff
@@ -118,3 +135,117 @@ def update_global_charges(request):
...
+
+# ================= Create New Record=========================
+@api_view(['POST'])
+@authentication_classes([JWTAuthentication])
+@validate_permission("obgyn_config","add")
+def create_obgyn_config(request):
+    data = {}
+    if request.method == "POST":
+        obgyn_config = ObgynConfigModel()
+        serializer = Obgyn_Configserializers(obgyn_config, data=request.data)
```
Comment (akhatri84) [inline obgyn_config/views.py:185]: Avoid using underscore in class names. Obgyn_Configserializers. Use ObgynConfigSerializers instead

File: obgyn_config/models.py (hunk 2 above)
Comment (akhatri84) [inline obgyn_config/models.py]: Change every foreign key names. remove _id from foreign keys as it is not required.

File: obgyn/urls.py (hunk 1 above)
Comment (akhatri84) [inline obgyn/urls.py]: Check the patients.urls.
include the subscription_purchase end point within subscription.urls.
main urls.py should have minimum end points.

File: obgyn_config/views.py [hunk 5]
```diff
@@ -118,3 +135,117 @@ def update_global_charges(request):
...
+    try:
+        if id:
+            obgyn_config = ObgynConfigModel.objects.filter(pk=id, deleted=0)
+        else:
+            obgyn_config = ObgynConfigModel.objects.filter(Q(created_by=1,deleted=0) | Q(created_by=request.data.get('created_by')))
+
+        data["total_record"] = len(obgyn_config)
+
+        obgyn_config, data = filtering_query(
+            obgyn_config, query_string, "obgyn_config_id", "OBGYN_CONFIG")
```
Comment (akhatri84) [inline obgyn_config/views.py]: You need to add OBGYN_CONFIG class in utility->filter_models.py to make filtering_query function to work.

File: subscription/models.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+class SubscriptionModel(models.Model):
+    subscription_id = models.AutoField(primary_key=True)
+    subscription_name = models.CharField(max_length=50)
+
+    actual_price = models.IntegerField()
+    discount = models.IntegerField()
+    sell_price = models.IntegerField()
+    duration = models.IntegerField()
+    description = models.CharField(max_length=100)
+
```
Comment (akhatri84) [inline subscription/models.py:15]: assign default values to all columns else validation rules in serializers.py

File: subscription/serializers.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            duplicate_subscription = duplicate_subscription.filter(~Q(pk=self.instance.subscription_id)).first()
+        else:
+            duplicate_subscription = duplicate_subscription.first()
+
+        if duplicate_subscription != None:
+            raise serializers.ValidationError("subscription already exist.")
+            
+        data["sell_price"] = int(data["actual_price"]) * int(data["discount"]) /100 
+
+        return data
```
Comment (akhatri84) [inline subscription/serializers.py:31]: Confirm with Niravbhai about the subscription duplicate checking function in serializer

File: subscription/views.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        if request.method == "DELETE":
+            result = subscription.update(deleted=1)
+            data["success"] = True
+            data["msg"] = "Data deleted successfully."
+            data["deleted"] = result
+            return Response(data=data, status=status.HTTP_200_OK)
+
+    # ================= Create New Record=========================
+    def post(self, request):
+        data = {}
```
Comment (akhatri84) [inline subscription/views.py]: why post is still class based view

File: subscription_purchase/models.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
+from django.db import models
+from django.db import models
+from django.utils.timezone import now
+from user.models import User
+from subscription.models import SubscriptionModel
+
+
+class Subscription_purchaseModel(models.Model):
+
+    subscription_Purchase_id = models.AutoField(primary_key=True)
```
Comment (akhatri84) [inline subscription_purchase/models.py]: subscription_Purchase_id, name of all column should be in lower case

File: subscription_purchase/models.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    invoice_no = models.IntegerField(default=0)
+    inv_year = models.CharField(max_length=15, default="")
+
+    created_by = models.IntegerField(default=1, unique=False)
+    deleted = models.IntegerField(default=0, unique=False)
+    created_at = models.DateTimeField(default=now)
+
+
+    def __str__(self):
+        return f"({self.subscription_Purchase_id})"
```
Comment (akhatri84) [inline subscription_purchase/models.py]: same as above comment case sensitive

File: subscription_purchase/views.py [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+    try:
+        if id:
+            subscription_purchase = Subscription_purchaseModel.objects.filter(pk=id, deleted=0)
+        else:
+            subscription_purchase = Subscription_purchaseModel.objects.filter(Q(created_by=1,deleted=0) | Q(created_by=request.data.get('created_by')))
+
+        data["total_record"] = len(subscription_purchase)
+
+        subscription_purchase, data = filtering_query(
+            subscription_purchase, query_string, "subscription_Purchase_id", "SUBSCRIPTION_PURCHASE")
```
Comment (akhatri84) [inline subscription_purchase/views.py]: add class in utility->filter_models

File: obgyn_config/views.py (hunk 3 above)
Comment (DhruvRathod77) [reply obgyn_config/views.py]: unnecessary import

**Comment by DhruvRathod77** [comment]
changes Complete

**Review by akhatri84 — APPROVED** [review]

**Pull request closed without merging**